### PR TITLE
AISDK-53: Run integration tests on different environments based on branch name

### PR DESCRIPTION
**Pull request by kbridbur**
Example: https://travis-ci.org/revdotcom/revai-node-sdk/builds/580929727

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,31 @@
...
+        secure: "Wl6dUEMeU0O34XrsJE04CysZhb4o8d/2taFGAgEBGfIpeS9mlqQJKFOIeXbbBqZsdK0WKVLPpi6IINfnag8kcymyShfLZUvGxzOpqhIWL4TGDOa1YwwiirDgQhhuRob+V13BURNdCWDwaNMVpHvZW45ickTVVaz8m7h9FTlxrljkSa+8KPxBGDgsBzTpA9/xplHxj+UZpBAN1o3fbS+jmexp3QCdPBMax3Atuu1PA2Qh9hmjnXU1KuzwsIPJq985rcmHllbLeRKk90smPJuDui7a+n2u9ujNY0GbsERqXRh7U/wBMpqdGfcMzvBolzLfNOOjalO0mlzM9VgGDBkIvaIJE5HSf0bR0ah6C9KP97ZRSJ49WWPcpsLvBZDELHUn2pUysVx+OKtTwelBl8d5AiqpliUjLxStREvnM//mOEjO/9cM4Dr3Vqev88vmQr3QDdmoqVymTASRotBRiTpSu9lso9ODC5xSi+O+XgvJoRq9yWnbc9a1iW7oFJCLjzkdwHTP5T7Y3IU95/Z6ESfL534mZv2SG43+s0AzjJG5+RsAfuQYmMf6Q5lFgDVDmXo11Q/nBY5XUPPQZtQyEenJLV1uM/aEj302/d6nwA7PTjjKOi8NnwMkrywbxqbUvMe50yxap3rpW+3uI8cwVLJHG0oxUsMXvy+j/ZP5ijaL91U="
+    - stage: Prod Integration
+      script: npm run test
```
Comment (kbridbur) [inline .travis.yml]: These scripts will change to be whatever we run to run the integration tests once those are done

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,31 @@
...
-script:
-  - "npm run build"
-  - "npm run lint"
-  - "npm run test"
\ No newline at end of file
+
+script: npm run test
+
+jobs:
+  include:
```
Comment (kbridbur) [inline .travis.yml:9]: Defines a list of jobs to be executed after the main script

File: .travis.yml [hunk 3]
```diff
@@ -1,8 +1,31 @@
...
+    - stage: build
+      script: npm run build
+    - stage: lint
+      script: npm run lint
+    - stage: Test Integration
+      script: npm run test
+      if: branch =~ /^feature/
```
Comment (kbridbur) [inline .travis.yml:23]: stage is only run if the condition is satisfied

File: .travis.yml [hunk 4]
```diff
@@ -1,8 +1,31 @@
...
+        secure: "Wl6dUEMeU0O34XrsJE04CysZhb4o8d/2taFGAgEBGfIpeS9mlqQJKFOIeXbbBqZsdK0WKVLPpi6IINfnag8kcymyShfLZUvGxzOpqhIWL4TGDOa1YwwiirDgQhhuRob+V13BURNdCWDwaNMVpHvZW45ickTVVaz8m7h9FTlxrljkSa+8KPxBGDgsBzTpA9/xplHxj+UZpBAN1o3fbS+jmexp3QCdPBMax3Atuu1PA2Qh9hmjnXU1KuzwsIPJq985rcmHllbLeRKk90smPJuDui7a+n2u9ujNY0GbsERqXRh7U/wBMpqdGfcMzvBolzLfNOOjalO0mlzM9VgGDBkIvaIJE5HSf0bR0ah6C9KP97ZRSJ49WWPcpsLvBZDELHUn2pUysVx+OKtTwelBl8d5AiqpliUjLxStREvnM//mOEjO/9cM4Dr3Vqev88vmQr3QDdmoqVymTASRotBRiTpSu9lso9ODC5xSi+O+XgvJoRq9yWnbc9a1iW7oFJCLjzkdwHTP5T7Y3IU95/Z6ESfL534mZv2SG43+s0AzjJG5+RsAfuQYmMf6Q5lFgDVDmXo11Q/nBY5XUPPQZtQyEenJLV1uM/aEj302/d6nwA7PTjjKOi8NnwMkrywbxqbUvMe50yxap3rpW+3uI8cwVLJHG0oxUsMXvy+j/ZP5ijaL91U="
```
Comment (kbridbur) [inline .travis.yml]: Api key and base url encrypted using travis cli

File: .travis.yml [hunk 5]
```diff
@@ -1,8 +1,31 @@
...
+        - secure: "Wl6dUEMeU0O34XrsJE04CysZhb4o8d/2taFGAgEBGfIpeS9mlqQJKFOIeXbbBqZsdK0WKVLPpi6IINfnag8kcymyShfLZUvGxzOpqhIWL4TGDOa1YwwiirDgQhhuRob+V13BURNdCWDwaNMVpHvZW45ickTVVaz8m7h9FTlxrljkSa+8KPxBGDgsBzTpA9/xplHxj+UZpBAN1o3fbS+jmexp3QCdPBMax3Atuu1PA2Qh9hmjnXU1KuzwsIPJq985rcmHllbLeRKk90smPJuDui7a+n2u9ujNY0GbsERqXRh7U/wBMpqdGfcMzvBolzLfNOOjalO0mlzM9VgGDBkIvaIJE5HSf0bR0ah6C9KP97ZRSJ49WWPcpsLvBZDELHUn2pUysVx+OKtTwelBl8d5AiqpliUjLxStREvnM//mOEjO/9cM4Dr3Vqev88vmQr3QDdmoqVymTASRotBRiTpSu9lso9ODC5xSi+O+XgvJoRq9yWnbc9a1iW7oFJCLjzkdwHTP5T7Y3IU95/Z6ESfL534mZv2SG43+s0AzjJG5+RsAfuQYmMf6Q5lFgDVDmXo11Q/nBY5XUPPQZtQyEenJLV1uM/aEj302/d6nwA7PTjjKOi8NnwMkrywbxqbUvMe50yxap3rpW+3uI8cwVLJHG0oxUsMXvy+j/ZP5ijaL91U="
+    - stage: Prod Integration
+      script: npm run test
+      if: branch = master
```
Comment (amikofalvy) [inline .travis.yml]: As we discussed in slack, we don't use the master branch.

We should instead map such as the following:
```
if: tag =~ /*.*.*/ -> prod
if: branch = develop & tag != ~ /*.*.*/ -> stage
if: branch =~ /^feature/ -> test
```

**Review by amikofalvy — CHANGES_REQUESTED** [review]
see comments inline

File: .travis.yml [hunk 6]
```diff
@@ -1,10 +1,40 @@
...
+
+jobs:
+  include:
+    - stage: build
+      script: npm run build
+    - stage: lint
+      script: npm run lint
+    - stage: Test Integration
+      script:
+        - npm run build
```
Comment (kbridbur) [inline .travis.yml:20]: We unfortunately need to re-run build because travis doesn't keep anything in between stages

File: .travis.yml [hunk 7]
```diff
@@ -1,10 +1,44 @@
...
-  - "npm run integration-test"
+
+script: npm run unit-test
+
+jobs:
+  include:
+    - stage: build
+      script: npm run build
+      node_js: "8"
+    - stage: build
```
Comment (kbridbur) [inline .travis.yml:13]: travis requires separate stages for separate versions

**Review by amikofalvy — APPROVED** [review]
Looks good.

**Pull request merged**